internal/dao/sqlx/source_dao.go

```diff
@@ -0,0 +1,95 @@
+package sqlx
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/RHEnVision/provisioning-backend/internal/dao"
+	"github.com/RHEnVision/provisioning-backend/internal/db"
+	m "github.com/RHEnVision/provisioning-backend/internal/models"
+	"github.com/jmoiron/sqlx"
+)
+
+const (
+	create     = `INSERT INTO sources (source_id, name, org_id, auth_id) VALUES ($1, $2, $3, $4) RETURNING created_at`
+	getById    = `SELECT * FROM sources WHERE source_id = $1;`
+	list       = `SELECT * FROM sources ORDER BY id LIMIT 100 OFFSET $1`
+	getByOrgId = `SELECT * FROM sources WHERE org_id = $1 ORDER BY id LIMIT 100;`
+)
+
+type sourceDaoSqlx struct {
+	getById    *sqlx.Stmt
+	list       *sqlx.Stmt
+	create     *sqlx.Stmt
+	getByOrgId *sqlx.Stmt
+}
+
+var GetSourceDao func(ctx context.Context) (dao.SourceDao, error)
+
+func getSourceDao(ctx context.Context) (dao.SourceDao, error) {
+	var err error
+	daoImpl := sourceDaoSqlx{}
+
+	daoImpl.getById, err = db.DB.PreparexContext(ctx, getById)
+	if err != nil {
+		fmt.Println(err.Error())
+		return nil, err
+	}
+	daoImpl.list, err = db.DB.PreparexContext(ctx, list)
+	if err != nil {
+		fmt.Println(err.Error())
+		return nil, err
+	}
+	daoImpl.create, err = db.DB.PreparexContext(ctx, create)
+	if err != nil {
+		fmt.Println(err.Error())
+		return nil, err
+	}
+	daoImpl.getByOrgId, err = db.DB.PreparexContext(ctx, getByOrgId)
+	if err != nil {
+		fmt.Println(err.Error())
+		return nil, err
+	}
+	return &daoImpl, nil
+}
+
+func init() {
+	println("Importing dao/sqlx source")
+	GetSourceDao = getSourceDao
+}
+
+func (s *sourceDaoSqlx) Create(source *m.Source) error {
+	_, err := db.DB.Exec("INSERT INTO sources (source_id, name, org_id, auth_id) VALUES ($1, $2, $3, $4) RETURNING created_at", source.SourceID, source.Name, source.OrgID, source.AuthID)
+	//_, err := s.create.Exec(source)
```

Comment:
Is there a problem? I haven't tried insert yet, I guess it should work this way...

Reply:
See the 'main' branch for working INSERT example.